go/kbfs/libkbfs/prefetcher.go

```diff
@@ -184,9 +187,16 @@ func (p *blockPrefetcher) applyToPtrParentsRecursive(
 			panic(r)
 		}
 	}()
-	for pptr := range pre.parents[ptr.RefNonce] {
+	for pptr, ch := range pre.parents[ptr.RefNonce] {
+		parentDone := false
+		select {
+		case <-ch:
+			parentDone = true
+		default:
+		}
+
 		parent, ok := p.prefetches[pptr.ID]
-		if !ok {
+		if parentDone || !ok {
```

Comment:
Please document this PR's description here as a comment.

Reply:
Done, and I refactored that logic into a helper so I only had to put the comment in one place.